engine/runtime/src/main/java/org/enso/interpreter/node/callable/resolver/HostMethodCallNode.java

```diff
@@ -122,16 +121,23 @@ public static PolyglotCallType getPolyglotCallType(
       return PolyglotCallType.CONVERT_TO_TIME_ZONE;
     } else if (library.isString(self)) {
       return PolyglotCallType.CONVERT_TO_TEXT;
-    } else if (library.isMemberInvocable(self, methodName)) {
+    } else if (library.hasArrayElements(self)) {
```

Comment:
Add a benchmark covering this check, I guarantee it will be unacceptably slow. Uncached method resolver digs through hashmaps. Unacceptable in fast paths (which this should be, it's used in a guard!)

Reply:
I'll write new benchmark tomorrow.

Have you had a chance to check the [benchmark results](https://www.pivotaltracker.com/story/show/183000876/comments/233194239) - are they OK?

Reply:
Since [96ad6f4](https://github.com/enso-org/enso/pull/3667/commits/96ad6f4210d9b059dafe83fe2c7db111b0f95c50) there are four comparable benchmarks testing matrix of `[Vector, Array]` and _[enso, polyglot]_ combinations: One can execute them as:
```
enso.$ sbt --java-home /graalvm "benchOnly VectorBenchmarks"
VectorBenchmarks.averageOverArray               0.072 ± 0.011  ms/op
VectorBenchmarks.averageOverPolyglotArray   0.068 ± 0.002  ms/op
VectorBenchmarks.averageOverPolyglotVector 0.032 ± 0.001  ms/op
VectorBenchmarks.averageOverVector              0.032 ± 0.001  ms/op
```
e.g. operations over array are twice as slow. Time to speed them up!

Reply:
With [a3b9b6e](https://github.com/enso-org/enso/pull/3667/commits/a3b9b6e8bcb4fd4e6d59f236849703dfbcac0d25) we reach these performance numbers and also work correctly with
```
enso$ ./built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/bin/enso --run test/Tests/src/Main.enso
```
e.g. one can still access additional JavaScript and Python properties of arrays.